### PR TITLE
[#931] New courses option not in Course details view (connect #931)

### PR DESCRIPTION
**Pull request by MacNew**

### CODE REVIEW

File: src/app/courses/view-courses/courses-view.component.html [hunk 1]
```diff
@@ -7,6 +7,12 @@
   <mat-toolbar class="primary-color font-size-1">
     <span class="margin-lr-3"><h3>{{courseDetail.courseTitle}}</h3></span>
     <span class="toolbar-fill"></span>
+    <button mat-button class="margin-lr-3" (click)="coursesService.admitCourse(courseDetail._id, courseDetail.admission)">
+    <span *ngIf="!parent">
+        <span *ngIf = "courseDetail.admission; else admission" i18n>Resigin</span>
```
Comment (paulbert) [inline src/app/courses/view-courses/courses-view.component.html]: Please remove spaces around =

File: src/app/courses/courses.service.ts [hunk 2]
```diff
@@ -40,4 +62,52 @@ export class CoursesService {
...
+      });
+      if (myCourseIndex > -1) {
+        return { ...course, admission: true };
+      }
+      return { ...course, admission: false };
+    });
+  }
+
+  admitCourse(courseId, participate) {
+    console.log('CourseId', courseId);
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: Please remove the console logs from this function.

File: src/app/courses/courses.component.ts [hunk 3]
```diff
@@ -56,45 +57,33 @@ export class CoursesComponent implements OnInit, AfterViewInit {
...
 
   constructor(
     private couchService: CouchService,
     private dialog: MatDialog,
     private planetMessageService: PlanetMessageService,
     private router: Router,
     private route: ActivatedRoute,
-    private userService: UserService
+    private userService: UserService,
+    public coursesService: CoursesService
```
Comment (paulbert) [inline src/app/courses/courses.component.ts]: Do not make a service public. Expose the methods you need within component methods.

File: src/app/courses/view-courses/courses-view.component.ts [hunk 4]
```diff
@@ -42,15 +42,24 @@ export class CoursesViewComponent implements OnInit, OnDestroy {
...
+    .subscribe((params: ParamMap) => {
+      const courseId = params.get('id');
+      const getOpts: any = { courseIds: [ courseId ] };
+      if (this.parent) {
+        getOpts.opts = { domain: this.userService.getConfig().parentDomain };
+      }
+      this.coursesService.updateCourses(getOpts);
+      this.coursesService.requestCourse({ courseId: params.get('id'), forceLatest: true });
+    }, error => (error));
+    this.coursesService.courseUpdated$.pipe(takeUntil(this.onDestroy$))
```
Comment (paulbert) [inline src/app/courses/view-courses/courses-view.component.ts]: Please do not switch the order of these functions.  It's unnecessary and simply makes it harder to review.

File: src/app/courses/courses.service.ts [hunk 5]
```diff
@@ -11,17 +17,31 @@ export class CoursesService {
...
+  userShelf = this.userService.getUserShelf();
+  private onDestroy$ = new Subject<void>();
 
   constructor(
-    private couchService: CouchService
-  ) {}
+    private couchService: CouchService,
+    private planetMessageService: PlanetMessageService,
+    private userService: UserService,
+    private router: Router
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: Unused in service

File: src/app/courses/courses.service.ts [hunk 6]
```diff
@@ -1,6 +1,12 @@
 import { Injectable } from '@angular/core';
 import { CouchService } from '../shared/couchdb.service';
 import { Subject } from 'rxjs/Subject';
+import { PlanetMessageService } from '../shared/planet-message.service';
+import { UserService } from '../shared/user.service';
+import { Router } from '@angular/router';
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: Router is not used in service

File: src/app/courses/courses.service.ts [hunk 7]
```diff
@@ -1,6 +1,12 @@
...
+import { switchMap, catchError, map, takeUntil } from 'rxjs/operators';
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: switchMap, catchError, and map are not used

File: src/app/courses/courses.service.ts [hunk 8]
```diff
@@ -11,17 +17,31 @@ export class CoursesService {
...
+      });
+    }
 
   // Components call this to get details of one course.
   // If the id already matches what is stored on the service, return that.
   // Or will get new version if forceLatest set to true
   requestCourse({ courseId, forceLatest = false }, opts: any = {}) {
     if (!forceLatest && courseId === this.course._id) {
-      this.courseUpdated.next(this.course);
+      const courseArray = [];
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: These three lines can easily be written as one line

File: src/app/courses/courses.service.ts [hunk 9]
```diff
@@ -30,7 +50,9 @@ export class CoursesService {
   private getCourse(courseId: string, opts) {
     this.couchService.get('courses/' + courseId, opts).subscribe(course => {
       this.course = course;
-      this.courseUpdated.next(course);
+      const courseArray = [];
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: These three lines can easily be written as one line

**Comment by lmmrssa** [comment]
There has been lot of changes in course after this. So it might be good to close this issue.

**Pull request closed without merging**